marlowe-chain-sync/src/Language/Marlowe/Runtime/ChainSync/NodeClient.hs

```diff
@@ -166,7 +171,10 @@ pipelinedClient costModel maxCost changesVar getHeaderAtPoint getIntersectionPoi
       -> NumberedChainTip
       -> IO (ClientPipelinedStIdle 'Z NumberedCardanoBlock ChainPoint NumberedChainTip IO ())
     clientStIdle point nodeTip = do
-      clientTip <- fmap blockHeaderToBlockNo <$> runGetHeaderAtPoint getHeaderAtPoint point
+      mClientTip <- (fmap . fmap) blockHeaderToBlockNo <$> runGetHeaderAtPoint getHeaderAtPoint point
+      clientTip <- case mClientTip of
+        Nothing -> error $ "Failed to start chain sync client. Failed to load block number for chain point " <> show point
+        Just tip -> pure tip
```

Comment:
On line 185, is `50` an appropriate limit? It seems that would depend upon the PostgreSQL configuration.

Reply:
It actually depends on the network more - this is the maximum pipelining limit of the chain sync protocol requests, not the database writes. 50 is the value that db-sync uses for this